### PR TITLE
Task/scala 5: Sample code

### PR DESCRIPTION
**Pull request by genevacoder**
Sample code for SCALA-5

### CODE REVIEW

File: core-scala/src/test/scala/com/baeldung/scala/GenericsIntro.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.baeldung.scala
+
+object GenericsIntro {
+  case class Apple(name: String)
```
Comment (ashleyfrieze) [inline core-scala/src/test/scala/com/baeldung/scala/GenericsIntro.scala:4]: Is this ever used?

Comment (genevacoder) [reply core-scala/src/test/scala/com/baeldung/scala/GenericsIntro.scala:4]: Not in the code. It was mentioned in the article inside a sentence. So we should only have code that is used within the scala file(s), right? Thanks.

Comment (ashleyfrieze) [reply core-scala/src/test/scala/com/baeldung/scala/GenericsIntro.scala:4]: I don't think this adds much, so let's lose it.

File: core-scala/src/test/scala/com/baeldung/scala/GenericsIntro.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    case class MagicHat(magic: AnyRef)
+
+    def run() = {
+      val rabbitHat = MagicHat(Rabbit(2))
+      val rabbit: Rabbit = rabbitHat.magic.asInstanceOf[Rabbit]
+      println(rabbit.cuteness)
+    }
+  }
+
+  object GenericWay{
```
Comment (ashleyfrieze) [inline core-scala/src/test/scala/com/baeldung/scala/GenericsIntro.scala:21]: I notice an inconsistency with the use of `{` after the name. Sometimes there's a space and sometimes there isn't. Which is the standard? is it `Name {` like in Java?

File: core-scala/src/test/scala/com/baeldung/scala/GenericsIntro.scala (hunk 1 above)
Comment (ashleyfrieze) [reply core-scala/src/test/scala/com/baeldung/scala/GenericsIntro.scala:4]: I've changed my mind... Let's keep it, but let's provide the code that would crash in the AnyRef context. More about this in my review of the article, which is on its way.

File: core-scala/src/test/scala/com/baeldung/scala/GenericsIntro.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  case class Apple(name: String)
+
+  def main(args: Array[String]): Unit = {
+    NonGenericWay.run()
+    GenericWay.run()
+  }
+
+  object LongHandWay {
+    case class AppleMagicHat(magic: Apple)
```
Comment (ashleyfrieze) [inline core-scala/src/test/scala/com/baeldung/scala/GenericsIntro.scala:13]: To make the point do we need to include the `RabbitMagicHat` here too?

**Pull request closed without merging**